feat: Implement OTP Store and enable client to set OTP expiry

sitaram-kalluri opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines in CONTRIBUTING.md

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**
- Enable the client to pass the OTP expiry duration, beyond which the OTP becomes invalid.

**- How I did it**
- Update the verb syntax for "OTP" to include a "TTL" parameter, which represents the duration in seconds.
- Introduce a new class called "OTPStore" to store OTPs along with their expiration durations.
- Implement a periodic job configured to run at specified intervals, responsible for removing expired OTPs from the "OTPStore"
- Configure the duration for this job using the "otp"-> "gcDuration" setting in the config.yaml file. By default, it runs every 1 hour.

- Changes in OtpVerbHandler.dart

  - Replace the "ExpireCache" - a third party library which is used to manage OTPs, with an instance of the "OTPStore" class. Since "ExpiryCache" is no longer used, removed it from the pubspec.yaml
  - In the constructor of the "OtpVerbHandler" class, initialize both the "OTPStore" and the "gcDuration." These are used to manage OTPs and determine how often the job for removing expired OTPs should run.
  - Introduce a new parameter called "otpExpiryDuration," which represents the duration beyond which an OTP expires. If the client does not specify a value for "TTL," it defaults to 5 minutes.
  - Implement the "isOTPValid" method, which returns a boolean value indicating whether an OTP is valid or not.
 
**- How to verify it**
Following are the unit tests:
 - A test to verify otp:get with TTL set is active before TTL is met
 - A test to verify otp:get with TTL set expires after the TTL is met
 - A test to verify gc removes the expired keys from the otp store

Following are the functional tests:
 - A test to generate OTP and returns valid before OTP is not expired
 - A test to generate OTP and returns invalid when TTL is met

 - Because the OTPs cannot be reused, updated the functional tests in enroll_verb_test.dart to fetch otp for each request

**- Description for the changelog**
- Enable clients to configure OTPs expiry duration
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->

sitaram-kalluri committed:
fix: Refactor commit log keystore and unit tests

sitaram-kalluri committed:
feat: Implement OTP Store and OTP expiry

sitaram-kalluri committed:
fix: Move getChanges method from "AtCommitLog" to "ClientAtCommitLog" and update the unit tests

sitaram-kalluri committed:
fix: Update at_commons package version

sitaram-kalluri committed:
fix: Update at_commons package version

dependabot[bot] committed:
build(deps): Bump ossf/scorecard-action from 2.2.0 to 2.3.0

Bumps [ossf/scorecard-action](https://github.com/ossf/scorecard-action) from 2.2.0 to 2.3.0.
- [Release notes](https://github.com/ossf/scorecard-action/releases)
- [Changelog](https://github.com/ossf/scorecard-action/blob/main/RELEASE.md)
- [Commits](https://github.com/ossf/scorecard-action/compare/08b4669551908b1024bb425080c797723083c031...483ef80eb98fb506c348f7d62e28055e49fe2398)

---
updated-dependencies:
- dependency-name: ossf/scorecard-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

dependabot[bot] committed:
build(deps): Bump github/codeql-action from 2.21.9 to 2.22.0

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 2.21.9 to 2.22.0.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/ddccb873888234080b77e9bc2d4764d5ccaaccf9...2cb752a87e96af96708ab57187ab6372ee1973ab)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

gkc reviewed (changes requested):
Since OTP lifespans can extend to weeks rather than just minutes or seconds, the OTPs need to persist through server restarts.

dependabot[bot] committed:
build(deps): Bump github/codeql-action from 2.22.0 to 2.22.1

Bumps [github/codeql-action](https://github.com/github/codeql-action) from 2.22.0 to 2.22.1.
- [Release notes](https://github.com/github/codeql-action/releases)
- [Changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md)
- [Commits](https://github.com/github/codeql-action/compare/2cb752a87e96af96708ab57187ab6372ee1973ab...fdcae64e1484d349b3366718cdfef3d404390e85)

---
updated-dependencies:
- dependency-name: github/codeql-action
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

sitaram-kalluri committed:
fix: Update the CHANGELOG.md

sitaram-kalluri committed:
fix: Remove OTPStore and persist OTPs in hive keystore

sitaram-kalluri committed:
Merge branch 'trunk' into expire_otp_changes

sitaram-kalluri committed:
fix: failing functional test

sitaram-kalluri committed:
Merge branch 'trunk' into expire_otp_changes

sitaram-kalluri committed:
fix: failing functional test

sitaram-kalluri commented:
Closing this PR. The changes are pushed as a part of the following PR: https://github.com/atsign-foundation/at_server/pull/1609
